### PR TITLE
Make preferred company face loading safe

### PR DESCRIPTION
**Pull request by duncanspumpkin**
This fixes the preferred company/currency selection being safe.

### CODE REVIEW

File: src/OpenLoco/src/Scenario.cpp [hunk 1]
```diff
@@ -531,47 +525,40 @@ namespace OpenLoco::Scenario
             }
             ObjectManager::selectObjectFromIndex(ObjectManager::SelectObjectModes::defaultDeselect, *oldCurrency.object._header, getInUseSelectedObjectFlags(), _objectSelectionMeta);
         }
-        ObjectManager::selectObjectFromIndex(ObjectManager::SelectObjectModes::defaultSelect, preferredCurreny, getInUseSelectedObjectFlags(), _objectSelectionMeta);
+        if (!ObjectManager::selectObjectFromIndex(ObjectManager::SelectObjectModes::defaultSelect, preferredCurreny, getInUseSelectedObjectFlags(), _objectSelectionMeta))
+        {
+            // Failed so reselect the old currency and give up
+            ObjectManager::selectObjectFromIndex(ObjectManager::SelectObjectModes::defaultSelect, *oldCurrency.object._header, getInUseSelectedObjectFlags(), _objectSelectionMeta);
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Scenario.cpp:531]: Should we unset the currency preferences at this point too?

File: src/OpenLoco/src/World/CompanyManager.cpp [hunk 2]
```diff
@@ -336,6 +336,11 @@ namespace OpenLoco::CompanyManager
         }
 
         auto competitorId = selectNewCompetitor(preferredOwner);
+        // This might happen if a preferredOwner object header does not exist anymore.
+        if (competitorId == kNullObjectId)
+        {
+            competitorId = selectNewCompetitor(nullptr);
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/World/CompanyManager.cpp:342]: Should we unset the competitor preferences/config at this point too?

File: src/OpenLoco/src/Scenario.cpp (hunk 1 above)
Comment (duncanspumpkin) [reply src/OpenLoco/src/Scenario.cpp:531]: I was thinking perhaps not. If you are just switching out objects for one time i think you would want it to return back when you have your objects

Comment (duncanspumpkin) [reply src/OpenLoco/src/Scenario.cpp:531]: But if you think that doesn't make sense then can change it.

Comment (AaronVanGeffen) [reply src/OpenLoco/src/Scenario.cpp:531]: Sounds sensible for that use case. I'm just worried it'll have unintended side effects for the company face selection window, for example. Is that working well like this?

Comment (duncanspumpkin) [reply src/OpenLoco/src/Scenario.cpp:531]: It will get unset if you open the options window whilst in this state. Same goes for currency.

Comment (AaronVanGeffen) [reply src/OpenLoco/src/Scenario.cpp:531]: Alright, let's leave it as-is then.

**Review by AaronVanGeffen — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 24.07+ (???)
 ------------------------------------------------------------------------
+- Fix: [#2555] Crash when setting preferred currency or preferred company to an object that no longer exists.
```
Comment (AaronVanGeffen) [inline CHANGELOG.md:3]: Could be split into two lines to include #1668.

**Pull request merged**